infrastructure/microvm/cloudhypervisor/create.go

```diff
@@ -0,0 +1,198 @@
+		return nil, err
+	}
+
+	cmd := exec.Command(p.config.CloudHypervisorBin, args...)
+
+	stdOutFile, err := p.fs.OpenFile(state.StdoutPath(), os.O_WRONLY|os.O_CREATE|os.O_APPEND, defaults.DataFilePerm)
+	if err != nil {
+		return nil, fmt.Errorf("opening stdout file %s: %w", state.StdoutPath(), err)
+	}
+
+	stdErrFile, err := p.fs.OpenFile(state.StderrPath(), os.O_WRONLY|os.O_CREATE|os.O_APPEND, defaults.DataFilePerm)
+	if err != nil {
+		return nil, fmt.Errorf("opening sterr file %s: %w", state.StderrPath(), err)
+	}
+
+	cmd.Stderr = stdErrFile
+	cmd.Stdout = stdOutFile
+	cmd.Stdin = &bytes.Buffer{}
+
+	var startErr error
+	if detached {
+		startErr = process.DetachedStart(cmd)
+	} else {
+		startErr = cmd.Start()
+	}
+
+	if startErr != nil {
+		return nil, fmt.Errorf("starting cloud hypervisor process: %w", err)
+	}
+
+	return cmd.Process, nil
+}
+
+func (p *provider) buildArgs(vm *models.MicroVM, state State, logger *logrus.Entry) ([]string, error) {
+	args := []string{
+		"--api-socket",
+		state.SockPath(),
+		"--log-file",
+		state.LogPath(),
+		"-v",
+	}
+
+	// Kernel and cmdline args
+	kernelCmdLine := DefaultKernelCmdLine()
+
+	for key, value := range vm.Spec.Kernel.CmdLine {
+		kernelCmdLine.Set(key, value)
+	}
+
+	args = append(args, "--cmdline", kernelCmdLine.String())
+	args = append(args, "--kernel", fmt.Sprintf("%s/%s", vm.Status.KernelMount.Source, vm.Spec.Kernel.Filename))
+
+	// CPU and memory
+	args = append(args, "--cpus", fmt.Sprintf("boot=%d", vm.Spec.VCPU))
+	args = append(args, "--memory", fmt.Sprintf("size=%dM", vm.Spec.MemoryInMb))
+
+	// Volumes (root, additional, metadata)
+	rootVolumeStatus, volumeStatusFound := vm.Status.Volumes[vm.Spec.RootVolume.ID]
+	if !volumeStatusFound {
+		return nil, cerrors.NewVolumeNotMounted(vm.Spec.RootVolume.ID)
+	}
+	args = append(args, "--disk", fmt.Sprintf("path=%s", rootVolumeStatus.Mount.Source))
+	args = append(args, fmt.Sprintf("path=%s,readonly=on", state.CloudInitImage()))
+
+	for _, vol := range vm.Spec.AdditionalVolumes {
+		status, ok := vm.Status.Volumes[vol.ID]
+		if !ok {
+			return nil, cerrors.NewVolumeNotMounted(vol.ID)
+		}
+		args = append(args, fmt.Sprintf("path=%s", status.Mount.Source))
+	}
+
+	// Network interfaces
+	for i := range vm.Spec.NetworkInterfaces {
+		iface := vm.Spec.NetworkInterfaces[i]
+
+		status, ok := vm.Status.NetworkInterfaces[iface.GuestDeviceName]
+		if !ok {
+			return nil, cerrors.NewNetworkInterfaceStatusMissing(iface.GuestDeviceName)
+		}
+
+		args = append(args, "--net")
+		if iface.Type == models.IfaceTypeMacvtap {
+			arg, err := p.createMacvtapArg(&iface, status)
+			if err != nil {
+				return nil, fmt.Errorf("creating macvtap interface: %w", err)
+			}
+			args = append(args, arg)
+		} else if iface.Type == models.IfaceTypeTap {
+			args = append(args, fmt.Sprintf("tap=%s,mac=%s", status.HostDeviceName, iface.GuestMAC))
+		} else {
+			logger.Warn("unknown network interface type", "name", iface.GuestDeviceName, "type", iface.Type)
```

Comment:
It will still end up with a command where the outcome is not predictable as the next parameter (if there's any) will be used as "dev" argument.

```
cmd --api-socket /foo --log-file /bar -v --net --other baz
```
In this case it will just cry "--net" has no argument but it's expected OR it will try to use `--other` as argument and will cry "i don't know what to do with baz".